runelite-api/src/main/java/net/runelite/api/Player.java

```diff
@@ -53,4 +57,70 @@ public PlayerComposition getPlayerComposition()
 	{
 		return new PlayerComposition(player.getComposition());
 	}
+
+	public Model getModel()
+	{
+		return player.getModel();
+	}
+
+	public Polygon[] getPolygons()
+	{
+		ArrayList<Polygon> polys = new ArrayList<>();
+		Model model = getModel();
+		int localX = player.getX();
+		int localY = player.getY();
+
+		// models are orientated north (1024) and there are 2048 angles total
+		int orientation = (player.getOrientation() + 1024) % 2048;
+
+		if (model == null)
+		{
+			return null;
+		}
+
+		int[] verticesX = model.getVerticesX().clone();
+		int[] verticesY = model.getVerticesY();
+		int[] verticesZ = model.getVerticesZ().clone();
+
+		int[] trianglesX = model.getTrianglesX();
+		int[] trianglesY = model.getTrianglesY();
+		int[] trianglesZ = model.getTrianglesZ();
+
+		if (orientation != 0)
+		{
+			setOrientation(model, orientation, verticesX, verticesZ);
+		}
+
+		for (int i = 0; i < trianglesX.length; i++)
+		{
+			Point x = Perspective.worldToCanvas(client, localX - verticesX[trianglesX[i]], localY - verticesZ[trianglesX[i]], -verticesY[trianglesX[i]]);
+			Point y = Perspective.worldToCanvas(client, localX - verticesX[trianglesY[i]], localY - verticesZ[trianglesY[i]], -verticesY[trianglesY[i]]);
+			Point z = Perspective.worldToCanvas(client, localX - verticesX[trianglesZ[i]], localY - verticesZ[trianglesZ[i]], -verticesY[trianglesZ[i]]);
+
+			int xx[] = {
+					x.getX(), y.getX(), z.getX()
+			};
+			int yy[] = {
+					x.getY(), y.getY(), z.getY()
+			};
+			polys.add(new Polygon(xx, yy, 3));
+		}
+
+		return polys.toArray(new Polygon[polys.size()]);
+	}
+
+	private void setOrientation(Model model, int orientation, int[] verticesX, int[] verticesZ)
+	{
+		int sin = Perspective.SINE[orientation];
+		int cos = Perspective.COSINE[orientation];
+
+		int[] originalX = model.getVerticesX();
```

Comment:
I think capturing the original verticiesX/Y is needed as the cloned references are modified